### PR TITLE
feat(framework) Introduce `try`-`except` block to `flwr run`

### PR DESCRIPTION
**Pull request by chongshenng**
Needed to catch exceptions and return a JSON-formatted error message.

Merge after
* #4620

### CODE REVIEW

File: src/py/flwr/cli/run/run.py [hunk 1]
```diff
@@ -87,19 +87,23 @@ def run(
...
-        _run_with_exec_api(app, federation_config, config_overrides, stream)
-    else:
-        _run_without_exec_api(app, federation_config, config_overrides, federation)
+        if "address" in federation_config:
+            _run_with_exec_api(app, federation_config, config_overrides, stream)
+        else:
+            _run_without_exec_api(app, federation_config, config_overrides, federation)
+    # pylint: disable=broad-except, unused-variable
+    except (typer.Exit, SystemExit, Exception):
+        _print_json_error()
```
Comment (jafermarq) [inline src/py/flwr/cli/run/run.py:106]: We'll be always calling this function? or should it be called only when the `JSON` CLI outpus flag is passed?

Comment (jafermarq) [reply src/py/flwr/cli/run/run.py:106]: nervermind, i see the logic in here will call it when needed only: https://github.com/adap/flower/pull/4619

File: src/py/flwr/cli/run/run.py [hunk 2]
```diff
@@ -198,3 +202,7 @@ def _run_without_exec_api(
         check=True,
         text=True,
     )
+
+
+def _print_json_error() -> None:
+    """Print error message as JSON."""
```
[the remark this answers is not in the document]
Comment (chongshenng) [reply src/py/flwr/cli/run/run.py:208]: This should already be here, will resolve for now and check the CI.

File: src/py/flwr/cli/run/run.py (hunk 1 above)
Comment (chongshenng) [reply src/py/flwr/cli/run/run.py:106]: Yes!

**Review by danieljanes — APPROVED** [review]

**Pull request merged**